nemo/core/neural_types/neural_type.py

```diff
@@ -112,6 +113,15 @@ def compare(self, second) -> NeuralTypeComparisonResult:
         else:
             return NeuralTypeComparisonResult.INCOMPATIBLE
 
+    def compare_and_raise_error(self, parent_type_name, port_name, second_object):
+        """ Method compares definition of one type with another and raises an error if not compatible. """
```

Comment:
what is the difference between "type difinition" and just type?

Reply:
in this case - no difference